cmd/dagger/module.go

```diff
@@ -69,6 +71,8 @@ func init() {
 	moduleInitCmd.Flags().StringVar(&moduleName, "name", "", "Name of the new module (defaults to parent directory name)")
 	moduleInitCmd.Flags().StringVar(&moduleSourcePath, "source", "", "Source directory used by the installed SDK. Defaults to module root")
 	moduleInitCmd.Flags().StringVar(&licenseID, "license", defaultLicense, "License identifier to generate. See https://spdx.org/licenses/")
+	moduleInitCmd.Flags().BoolVar(&mergeDeps, "merge", false, "Merge module dependencies with existing project ones")
+	moduleInitCmd.Flags().MarkHidden("merge")
```

Comment:
Hm, I'm actually not sure about this option at all. (It's also a bit weird to have a go sdk specific flag right here, if we do want to have it, we probably want to make it a bit more generic or hide it if `--sdk=go` isn't set).

I think we should probably default to always generating `go.mod`/`go.sum`. If the user doesn't want them, they can delete them after, right? As long as we have a test for that, and check it works as expected and gets merged into the root, I think that's fine.

That way, go is like all the other sdks, which manage their own dependencies.

Reply:
The feedback about allowing backwards compatibility with a flag came from @shykes https://discord.com/channels/707636530424053791/1003718839739105300/1266140398112346203. 

I'd have preferred to avoid it since it makes the implementation cleaner but on the other hand I understand the reason to add it also.

Reply:
> if we do want to have it, we probably want to make it a bit more generic or hide it if --sdk=go isn't set).

The flag is hidden by default, so users won't know it's there. It's mostly a "safeguard" in case someone is trying to get the current "merge by default" behavior. IMO it's very unlikely to happen, but if it does, we can tell them to use this hidden flag.

Reply:
As long as the flag is hidden, we can wait for someone to ask for it, and if nobody does, we'll just remove it.